### ISSUE TITLE
n3DS: Loading content sometimes shows wrong current core

### ISSUE THREAD

**Issue by bparker06** [libretro/RetroArch#4888]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

[Description of the bug]

### Expected behavior

[What you expected to happen]

### Actual behavior

When changing content for the currently loaded core, sometimes the previously launched core shows up as the "current core" in the selection list.

### Steps to reproduce the bug

1. Start any core/content and then close RetroArch
2. Start new/different core/content from 3DS main menu
3. Try to load new content on same core that is already loaded
4. Current core shows up as the previously launched core

### Version/Commit
You can find this information under Information/System Information

- RetroArch: Apr 11 2017 nightly (no git version is provided)

### Environment information

- OS: n3DS
- Compiler: GCC 5.3.0

**Comment by Ferk** [comment]
Yes, this happens for me every time on an n3DS XL, if the content is running in the background it will use the same core.
It does open in the correct core if I close the content first from the quickmenu. It's the workaround I'm currently using.

I've also had it happen (only once) in Windows, when I was testing the "Craft" core and I opened a NES game without closing Craft.
